app/src/main/java/com/quarkworks/apartmentgroceries/MainActivity.java

```diff
@@ -18,14 +18,17 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.main_activity);
         SharedPreferences sharedPreferences = getSharedPreferences(getString(R.string.login_or_sign_up_session), 0);
-        String sessionToken = sharedPreferences.getString(SyncUser.JsonKeys.GROUP_ID, null);
+        String sessionToken = sharedPreferences.getString(RUser.JsonKeys.SESSION_TOKEN, null);
 
-        Intent intent;
         if (TextUtils.isEmpty(sessionToken)) {
-            intent = new Intent(MainActivity.this, LoginActivity.class);
+            LoginActivity.newIntent(this);
         } else {
-            intent = new Intent(MainActivity.this, HomeActivity.class);
+            ((MyApplication) MyApplication.getContext()).setSessionToken(sessionToken);
+            String userId = sharedPreferences.getString(RUser.JsonKeys.USER_ID, null);
+            String groupId = sharedPreferences.getString(RGroup.JsonKeys.GROUP_ID, null);
+            ((MyApplication) MyApplication.getContext()).setUserId(userId);
```

Comment:
In MyApplication, instead of holding on to an instance of Context, hold on to an instance of MyApplication and use .getInstance() to return it. Application is a subclass of Context, so you can use that directly as the Context.